File: services/access-control-service/src/main/java/com/omvaultchain/accesscontrol/model/AccessPermission.java
package com.omvaultchain.accesscontrol.model;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Data;

import java.time.Instant;

@Entity
@Table(name = "access_permission")
@Data
public class AccessPermission {
    @Id
    private String id;

    private String fileId;
    private String userId;
    private String encryptedKey;
    private Instant grantedAt;
    private Instant revokedAt;
    private Boolean isActive;
}

File: backend/app/Models/AccessPermission.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Concerns\HasUuids;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class AccessPermission extends Model
{
    use HasUuids;
    protected $table = 'access_permissions';
    public $incrementing = false;
    protected $keyType = 'string';
    protected $fillable = [
        "id",
        "file_id",
        "user_id",
        "encrypted_key",
        "granted_at",
        "revoked_at",
        "is_active",
    ];
    protected $casts = [
        'is_active' => 'boolean',
        'granted_at' => 'datetime',
        'revoked_at' => 'datetime',
    ];
    public function file(): BelongsTo{
        return $this->belongsTo(File::class, 'file_id');
    }
    public function user(): BelongsTo{
        return $this->belongsTo(User::class, 'user_id');
    }
    public function scopeActive($query){
        return $query->where('is_active', true)->whereNull('revoked_at');
    }
}
